### ISSUE TITLE
MetaGSCA

### ISSUE THREAD

**Issue by Haocan223** [Bioconductor/Contributions#1568]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/Haocan223/MetaGSCA

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by mtmorgan** [comment]
Your package must have version 0.99.0 to begin the review process; please update and post a comment here when ready. Please review package and submission guidelines to ensure other requirements are met.

**Comment by mtmorgan** [comment]
Sorry, your package now has files
```
  /inst/extdata/KIRP.csv
  /inst/extdata/LIHC.csv
```
greater than 5MB; these cannot be added to our git repository. If these are meant as examples, then perhaps it is appropriate to create smaller versions of the files. If they cannot be made smaller, consider creating an ExperimentHub or AnnotationHub package, e.g., https://bioconductor.org/packages/devel/bioc/vignettes/ExperimentHub/inst/doc/CreateAnExperimentHubPackage.html

**Comment by Haocan223** [comment]
Hi Martin,
We've changed the version to 0.99.0 and other requirement based on Bioconductor Package Guidelines. And I'll work on the KIRP.csv and LIHC.csv file now. Thanks so much for your feedback!
Best wishes,
Haocan

**Comment by Haocan223** [comment]
Hi Martin, I transform the "KIRP.csv & LIHC.csv" to "KIRP.rds & LIHC.rds". Both files are smaller than 5 MB. Hope this could be added to the git repository. Thanks in advance!!

**Comment by nturaga** [comment]
This is fixed now.

https://github.com/Bioconductor/repo-specific/commit/f1d4a20b74f2c72578bc50f9e7d5559020655858

**Comment by nturaga** [comment]
Hi, it seems your package is exceeding 5MB limit for Bioconductor software packages. Please fix this. 

```
    * ERROR: Package Source tarball exceeds Bioconductor size
      requirement.
        Package Size: 18.3502 MB
        Size Requirement: 5.0000 MB
```

Also, make sure your build report is as clean as possible.  Especially, 

1. 

```
    * NOTE: Avoid 1:...; use seq_len() or seq_along()
      Found in files:
        dichotGSAR.R (line 10, column 24)
        MetaGSCA.R (line 80, column 17)
        MetaGSCA.R (line 84, column 17)
        MetaGSCA.R (line 88, column 24)
        MetaGSCA.R (line 101, column 32)
        MetaGSCA.R (line 126, column 15)
        MetaGSCA.R (line 161, column 21)
        MetaGSCA.R (line 165, column 21)
        MetaGSCA.R (line 169, column 28)
        MetaGSCA.R (line 181, column 36)
        MetaGSCA.R (line 311, column 15)
        MetaGSCA.R (line 326, column 15)
        MetaGSCA.R (line 351, column 19)
        MetaGSCA.R (line 353, column 22)
        MetaGSCA.R (line 451, column 19)
        MetaGSCA.R (line 453, column 22)
        MetaGSCA.R (line 596, column 15)
        PWcTalkNWpre.R (line 10, column 39)
        PWcTalkNWpre.R (line 12, column 47)
        PWsim.R (line 16, column 13)
```

2. 

```
    * WARNING: Use TRUE/FALSE instead of T/F
      Found in R/ directory functions:
        dichotGSAR()
        GSAR_boot()
        PWcTalkNW()
```

Go through the build report and fix these. The review will be conducted once the build report is clean.

**Comment by Haocan223** [comment]
Really appreciate for your feedback! We've successfully figured the: 1 error and 2 warning messages. The package is now less than 5M. Hope it will work this time. Thanks!

**Comment by lshep** [comment]
Please increment the version number in the description and push to git.bioconductor.org to trigger a new build reflecting these changes before nitesh will re-review. thank you

**Comment by Haocan223** [comment]
Hi this is Haocan. Thanks so much for your detailed feedback. I checked the version number instruction on: https://bioconductor.org/developers/how-to/version-numbering/ and changed the version from 0.99.0 to 0.99.1. Please let me know whether this works! Thanks!!

**Comment by lshep** [comment]
You need to push it to git.bioconductor.org.  Instructions on how to do so can be found [here](https://bioconductor.org/developers/how-to/git/new-package-workflow/)

**Comment by nturaga** [comment]
Hi @Haocan223 

There is no updated build report. You need to update the version number in the DESCRIPTION file so the reviewer (me) can take a look at the latest build report. That is the prerequisite for packages to be reviewed.

**Comment by nturaga** [comment]
# MetaGSCA review

## DESCRIPTION

- use `Authors@R` instead of `Author` and `Maintainer` fields. 

	look at http://bioconductor.org/developers/package-guidelines/#description
	
## NAMESPACE

- ok

- Consider using roxygen2 for generating the NAMESPACE automatically.

## R 

- remove commented out lines of code.

- use the package `matrixStats` to calcualte row-wise and column-wise
  statistics like `sd` and `mean` etc. Look up the package. Do not use
  `apply` functions.
  
- Make your functions modular and more readable. If you write a
  function of length 300 lines, it'll be extremely fragile and you
  will find it hard to debug. Example the MetaGSCA.R function.
  
- Check your indentation in the R code. Take a look at
  http://bioconductor.org/developers/how-to/coding-style/. Follow
  it. It is very hard to follow and review your code as it is written.
  
- Do not `write.csv` without the users permission. This is bad coding practice.

- Make sure the end user messages are following this convention,

```
End-User messages

message() communicates diagnostic messages (e.g., progress during lengthy computations) during code evaluation.
warning() communicates unusual situations handled by your code.
stop() indicates an error condition.
cat() or print() are used only when displaying an object to the user, e.g., in a show method.
```

- Your package will not move forward without getting a build report.

## inst/extdata

- Your inst/extdata folder has many plots and png files and lsts. Can you explain why?

## Vignette

- Your vignette seems incomplete. Please fix.

- The code chunks should avoid using `eval=FALSE`.

**Comment by nturaga** [comment]
Hi @Haocan223 ,

We usually give the package authors/maintainers about 14 days to have responded to the comments we issue on the review. 

Please make sure you try to address these in time. If not we can always close the issue and take a look at it again. There is no problem to reopen the issue when you have more time to address the concerns. 

Please check the release schedule we are very close to it. http://bioconductor.org/developers/release-schedule/
